Library.java

```diff
@@ -0,0 +1,261 @@
+                }
+                catch (Exception e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            else self = new Library();
+        }
+        return self;
+    }
+
+
+    public static synchronized void saveInstanceToFile() {
+        if (self != null) {
+            self.loanDate = Calendar.getInstance().getDate();
+            try (FileOutputStream libraryFileOutput = new FileOutputStream(LIBRARY_FILE);
+                    ObjectOutputStream libraryObjectOutput = new ObjectOutputStream(libraryFileOutput);) {
+                libraryObjectOutput.writeObject(self);
+                libraryObjectOutput.flush();
+                libraryObjectOutput.close();
+            }
+            catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+
+    public int getBookId() {
+        return bookId;
+    }
+
+
+    public int getMemberId() {
+        return memberId;
+    }
+
+
+    private int getNextBookId() {
+        return bookId++;
+    }
+
+
+    private int getNextMemberId() {
+        return memberId++;
+    }
+
+
+    private int getNextLoanId() {
+        return loanId++;
+    }
+
+
+    public List<member> getMemberList() {
+        Collection<member> membersInMap = members.values();
+        return new ArrayList<member>(membersInMap);
+    }
+
+
+    public List<Book> getBookList() {
+        Collection<Book> booksInMap = catalog.values();
+        return new ArrayList<Book>(booksInMap);
+    }
+
+
+    public List<loan> getCurrentLoansList() {
+        Collection<loan> loansInMap = currentLoans.values();
+        return new ArrayList<loan>(loansInMap);
+    }
+
+
+    public member addMember(String lastName, String firstName, String email, int phoneNo) {
+        int nextMemberId = this.getNextMemberId();
+        member newMember = new member(lastName, firstName, email, phoneNo, nextMemberId);
+        members.put(newMember.GeT_ID(), newMember);
+        return newMember;
+    }
+
+
+    public Book addBook(String a, String t, String c) {
+        int nextBookId = this.getNextBookId();
+        Book newBook = new Book(a, t, c, nextBookId);
+        catalog.put(newBook.getBookId(), newBook);
+        return newBook;
+    }
+
+
+    public member getMemberId(int memberId) {
+        if (members.containsKey(memberId))
+            return members.get(memberId);
+        return null;
+    }
+
+
+    public Book getBookId(int bookId) {
+        if (catalog.containsKey(bookId))
+            return catalog.get(bookId);
+        return null;
+    }
+
+
+    public int getLoanLimit() {
+        return LOAN_LIMIT;
+    }
+
+
+    public boolean canMemberBorrow(member currentMember) {
+        if (currentMember.Number_Of_Current_Loans() == LOAN_LIMIT)
+            return false;
+
+        if (currentMember.Fines_OwEd() >= MAX_FINES_OWED)
+            return false;
+
+        for (loan loanBeingAssessed : currentMember.GeT_LoAnS())

```

Comment:
All method names are to start with a lowercase letter and to be in camelBack

Reply:
Methods 'Fines_OwEd()' and 'GeT_LoAnS()' belong to member.java, not Library.java.